### PR TITLE
feat(core) Cache length of CATEGORIES

### PR DESCRIPTION
**Pull request by depay**
Signed-off-by: Wei.ZHAO <zhaowei@qiyi.com>

### CODE REVIEW

File: kong/core/router.lua [hunk 1]
```diff
@@ -530,7 +530,7 @@ function _M.new(apis)
         -- hosts/URIs/methods
         local category_idx = CATEGORIES_LOOKUP[req_category]
 
-        while category_idx <= #CATEGORIES_LOOKUP do
+        while category_idx <= #CATEGORIES do
```
Comment (thibaultcha) [inline kong/core/router.lua]: I think it'd be even better to cache this value at the top level of the module and use it instead:
```
local categories_len = #CATEGORIES
```

Comment (depay) [inline kong/core/router.lua]: :+1:

**Comment by depay** [comment]
updated

**Comment by thibaultcha** [comment]
Thanks! Merged with an updated commit message.

**Pull request merged**